### PR TITLE
Captain can buy a new escape shuttle

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
rscadd: The Captain can now purchase alternate escape shuttles from the communications console. This drains from the station supply of cargo points, and you can only do so once per round, so spend wisely.
rscadd: Some shuttles are less desirable than the default, and will instead grant the station a credit bonus when purchased.
/:cl:

### CODE REVIEW

**Comment by AdamElTablawy** [comment]
traitor hop is going to buy a supermatter shuttle every fucking round

**Comment by KorPhaeron** [comment]
@AdamElTablawy 

If he can get literally INFINITY credits then sure

**Comment by TehZombehz** [comment]
More ways to mess with everybody I guess.

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -130,6 +131,31 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 				message_admins("[key_name_admin(usr)] has sent a message to the other server.")
 				CM.lastTimeUsed = world.time
 
+		if("purchase_menu")
+			state = STATE_PURCHASE
+
+		if("buyshuttle")
+			if(authenticated==2)
+				var/datum/map_template/shuttle/S = locate(href_list["chosen_shuttle"])
```
Comment (Iamgoofball) [inline code/game/machinery/computer/communications.dm]: hrefsploits

Comment (KorPhaeron) [inline code/game/machinery/computer/communications.dm]: tell me how to fix these things then

Comment (Iamgoofball) [inline code/game/machinery/computer/communications.dm]: ask @optimumtact and @MrStonedOne they're the experts on this meme

Comment (pubby) [inline code/game/machinery/computer/communications.dm]: looking at the changes made in #20732 this is probably all you need
```
var/list/shuttles = flatten_list(shuttle_templates)
var/datum/map_template/shuttle/S = locate(href_list["chosen_shuttle"]) in shuttles
if(S && istype(S))
    // ....
```

**Comment by Incoming5643** [comment]
Unless you've added extra checks you'd need to map edit most maps to be sure they had docking areas at least as large as the largest ship offered. If you order a ship too large for the dock it simply never comes and the round becomes unending.

**Comment by KorPhaeron** [comment]
@Incoming5643 

Does it give any kind of warning or logging that this is happening?

**Comment by KorPhaeron** [comment]
Also I thought giant shuttles could just crush things out of the way, what changed?

**Comment by KorPhaeron** [comment]
@Incoming5643 

I think I fixed it

**Comment by jughu** [comment]
Add the supermatter shuttle for 1 million credits :^)

**Comment by Supermichael777** [comment]
1000000 credits

**Comment by ExcessiveUseOfCobblestone** [comment]
Not sure if you fixed this in the PR as well, but pubbystation's shuttle uses a different area than all the other shuttles which could cause issues when trying to purchase shuttles [not to mention pubby is the only station that has escape on westwing which also causes issues].

May want to just disable the ability on pubby.

**Comment by KorPhaeron** [comment]
@ExcessiveUseOfCobblestone 

If the map is set up properly it will rotate the shuttles

**Comment by WJohn** [comment]
Finally a use for cargo points other than just cargo.

I wonder if we'll have an actual economy at some point used for other things, like vending machines, buying power when in need (to a radar dish of some kind) or beaming out excess to reward clever engineers.

**Comment by pubby** [comment]
Cobby is right this *will* break on pubbystation. The docking point on that map is defined backwards and floats 18 tiles to the left of the airlock. 

It's caused trouble though so I'll look into making it a normal dock.

**Comment by optimumtact** [comment]
Why have you done this pubby, you are tearing us all apart

**Comment by KorPhaeron** [comment]
I guess if people buy a new shuttle on pubby they'll have to jump for it

**Comment by Tacolizard** [comment]
ree pubby is so dank just fix yo map

**Comment by pubby** [comment]
don't worry I got this
#21405

**Comment by Tacolizard** [comment]
Just port the entirety of bay code at this point.

**Comment by A-t48** [comment]
Can I make an ice rink shuttle?

**Comment by kevinz000** [comment]
"Even if it has to crush a few departments in the process"

File: code/game/machinery/computer/communications.dm [hunk 2]
```diff
@@ -130,6 +131,32 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
...
 
+		if("purchase_menu")
+			state = STATE_PURCHASE
+
+		if("buyshuttle")
+			if(authenticated==2)
+				var/list/shuttles = flatten_list(shuttle_templates)
+				var/datum/map_template/shuttle/S = locate(href_list["chosen_shuttle"]) in shuttles
+				if(S && istype(S))
+					for(var/obj/machinery/shuttle_manipulator/M in machines)
```
Comment (optimumtact) [inline code/game/machinery/computer/communications.dm]: Huh?

There's only one shuttle manipulator isn't there, just store a reference to it in SSshuttle

come on kor this is bad and you know better than that.

File: code/datums/shuttles.dm [hunk 3]
```diff
@@ -88,14 +95,17 @@
 		crates!"
 	admin_notes = "Due to origin as a solo piloted secure vessel, has an \
 		active GPS onboard labeled STV5."
+	credit_cost = -7500
```
Comment (KorPhaeron) [inline code/datums/shuttles.dm]: Because it's tiny.

Comment (optimumtact) [inline code/datums/shuttles.dm]: Then why make it available at all

File: code/game/machinery/computer/communications.dm (hunk 2 above)
Comment (pubby) [inline code/game/machinery/computer/communications.dm]: Nah, don't blame him. I told him to put that code there.

File: code/datums/shuttles.dm (hunk 3 above)
Comment (KorPhaeron) [inline code/datums/shuttles.dm]: So you can get a shittier shuttle in exchange for a short term benefit, it's a trade off

File: code/game/machinery/computer/communications.dm (hunk 2 above)
Comment (Jordie0608) [inline code/game/machinery/computer/communications.dm]: Then you're both to blame.

**Comment by KorPhaeron** [comment]
@optimumtact 

It's been updated

**Comment by KorPhaeron** [comment]
@coiax 

:tada:

**Comment by coiax** [comment]
+1

**Comment by KorPhaeron** [comment]
@coiax 

Now get sucked back into coding and make some more fun shuttles please

**Pull request merged**